graylog2-web-interface/src/components/extractors/MessageLoader.tsx

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright (C) 2020 Graylog, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the Server Side Public License, version 1,
+ * as published by MongoDB, Inc.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+import * as React from 'react';
+import { useCallback, useEffect, useRef, useState } from 'react';
+import PropTypes from 'prop-types';
+
+import ActionsProvider from 'injection/ActionsProvider';
+
+import { Button } from '../graylog';
+
+const MessagesActions = ActionsProvider.getActions('Messages');
+
+type LoadMessageFormProps = {
+  loadMessage: (e: React.FormEvent) => void,
+  children: React.ReactNode,
+  loading: boolean,
+};
+const LoadMessageForm = ({ loadMessage, children, loading }: LoadMessageFormProps) => (
+  <div>
+    <form className="form-inline message-loader-form" onSubmit={loadMessage}>
+      {children}
+      <Button bsStyle="info"
+              disabled={loading}
+              type="submit">
+        {loading ? 'Loading message...' : 'Load message'}
+      </Button>
+    </form>
+  </div>
+);
+
+type Props = {
+  hidden: boolean,
+  hideText: string,
+  onMessageLoaded: (data: any) => void,
+  messageId: undefined | string,
+  index: undefined | string,
```

Comment:
Do we also want to include these new props in the prop types definition? I would also suggest setting their default values to `''`, so we don't see these warnings when editing the inputs:

```
Warning: A component is changing an uncontrolled input of type text to be controlled. Input elements should not switch from uncontrolled to controlled (or vice versa). Decide between using a controlled or uncontrolled input element for the lifetime of the component. More info: https://fb.me/react-controlled-components
```

Reply:
By the way, while playing with the component I noticed that with the current type definition, IntelliJ believes that both `messageId` and `index` are mandatory props:

![Screenshot 2021-05-17 at 12 20 04](https://user-images.githubusercontent.com/716185/118473693-8d1d5d80-b70a-11eb-8311-b59e69db5a61.png)


When I changed the type definition to the one below, the warning was gone and ESLint also complained that those optional props should have a default value, so I think that may be a slightly better option for setting optional props. What do you think?

```suggestion
  messageId?: string,
  index?: string,
```

Reply:
:white_check_mark: